src/app/projects/projlist-map/projlist-map.component.ts

```diff
@@ -289,6 +291,7 @@ export class ProjlistMapComponent implements AfterViewInit, OnChanges, OnDestroy
           const shapeurl = this.pathAPI + '/document/' + projectShapefile.shapefileId + '/fetch/' + escapedName;
           const shapefile = new L.Shapefile(shapeurl, { isArrayBufer: false, style: shapeFileStyle })
           .on('click', L.Util.bind(this.onShapefileClick, this, app));
+          shapefile.projectId = app._id;
```

Comment:
Just a thought, we could get rid of that pesky "app" language all together. Optional of course.